security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,111 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Failure Modes and Recovery Paths](#failure-modes-and-recovery-paths)
+  - [[Name of Failure Mode 1]](#name-of-failure-mode-1)
+  - [[Name of Failure Mode 2]](#name-of-failure-mode-2)
+- [Audit Requirements](#audit-requirements)
+- [Action Items](#action-items)
+- [Appendix](#appendix)
+  - [Appendix A: This is a Placeholder Title](#appendix-a-this-is-a-placeholder-title)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Stage 1.4: Kona and Asterisc FMA (Failure Modes and Recovery Path Analysis)
+
+| | |
+|--------|--------------|
+| Author | Andreas Bigger |
+| Created at | 2024-10-17 |
+| Initial Reviewers | *Reviewer Name 1, Reviewer Name 2* |
+| Need Approval From | Ben Clabby, Mark Tyneway |
+| Status | Draft |
+
+
+## Introduction
+
+This document covers the initial deployment of [the Asterisc fault proof VM](https://github.com/ethereum-optimism/asterisc) with [the Kona fault proof program](https://github.com/anton-rs/kona).
+
+Together, Kona and Asterisc form an alternative proof stack to the [op-program](https://github.com/ethereum-optimism/optimism/tree/develop/op-program) and [cannon](https://github.com/ethereum-optimism/optimism/tree/develop/cannon). A secondary dispute game will be created that uses kona and asterisc, with the `op-challenger` supporting playing _both_ game types - `op-program` + `cannon` as well as `kona` + `asterisc`.
+
+Since two main components are being shipped in Stage 1.4 there are two main places the system can break: the kona fault proof program, and the asterisc VM. The below Failure Modes and Recovery Paths go over those situations.
+
+Below are references for this project:
+
+- ["Stage 1.4" internal project document](https://www.notion.so/oplabs/Stage-1-4-Partway-a6f57ad777b148dda01488f2646cff17)
+- [Kona Documentation](https://anton-rs.github.io/kona/).
+- [Kona Repository](https://github.com/anton-rs/kona).
+- [Asterisc Repository](https://github.com/ethereum-optimism/asterisc)
+
+
+## Failure Modes and Recovery Paths
+
+
+### Divergent Kona Derivation Pipeline
+
+- **Description:** The Kona derivation pipeline diverged in behavior from the op-program's derivation pipeline.
+
+- **Risk Assessment:** Low likelihood, low impact. A small divergence in derivation can cause invalid game results. But since kona + asterisc is in the coldpath, it doesn't have impact on anything in the production system.
+
+- **Mitigations:** Frequent action tests covering the derivation pipeline.
```

Comment:
What exactly do you mean by frequent action tests? My understanding of action tests are the tests that we write to catch certain protocol behaviors/edge cases, they aren't fuzz tests that could catch net new behaviors

Reply:
Frequent here refers to them being frequently run in the kona repository since they are run on every pull request. I will update to be more specific.